core/src/test/scala/org/broadinstitute/dsde/rawls/dataaccess/HttpLeonardoDAOSpec.scala

```diff
@@ -0,0 +1,32 @@
+package org.broadinstitute.dsde.rawls.dataaccess
+
+import akka.actor.ActorSystem
+import akka.testkit.TestKit
+import com.typesafe.config.{Config, ConfigFactory}
+import org.broadinstitute.dsde.rawls.config.LeonardoConfig
+import org.broadinstitute.dsde.workbench.client.leonardo.ApiClient
+import org.broadinstitute.dsde.workbench.client.leonardo.api.AppsV2Api
+import org.scalatest.flatspec.AnyFlatSpecLike
+
+import java.util.UUID
+
+import scala.language.postfixOps
+
+class HttpLeonardoDAOSpec
+  extends TestKit(ActorSystem("HttpLeonardoDAOSpec"))
+    with AnyFlatSpecLike {
+
+  val apiClient = new ApiClient()
+
+  val workspaceId: UUID = UUID.randomUUID()
+
+  val testConf: Config = ConfigFactory.load()
+  val leonardoConfig: LeonardoConfig = LeonardoConfig.apply(testConf)
+
+  "HttpLeonardoDAO" should "get an AppsV2Api object when calling getAppsV2leonardoApi during app creation" in {
+    val dao = new MockLeonardoDAO(leonardoConfig.baseUrl, "CROMWELL")
+    val result = dao.getAppsV2leonardoApi("token")
+    assert(result.isInstanceOf[AppsV2Api])
+  }
+
```

Comment:
I decided to omit unit tests for other methods since they seem to specifically invoke the Leonardo Client (e.g. regressions should be captured in both contract tests and Leonardo Client tests)

Let me know if you disagree here -- I timeboxed doing simple tests for number of times invoked with no success (I know the code works manually, but the unit testing for it has been a much bigger engineering lift than reward here)

Reply:
The most valuable test in my opinion is the one you wrote in MultiCloudWorkspaceServiceSpec: when creating a workspace, assert that we call `LeonardoDAO. createWDSInstance()`.

I think you could do away with the test in this file, it's really only testing code inside the mock dao, and it's testing behavior of the client library, not really testing any of our own code.

Potentially you could write a unit test that validates that the WDS-specific `createWDSInstance()` method supplies `createApp()` with the right values, but that's also not a super high priority.